nomad/fsm.go

```diff
@@ -838,6 +838,7 @@ func (n *nomadFSM) handleJobDeregister(index uint64, jobID, namespace string, pu
 		}
 
 		stopped := current.Copy()
+		stopped.SetSubmitTime()
```

Comment:
FSM operations must be deterministic since all servers apply every Raft log and must reach the exact same state. Since `SetSubmitTime()` calls `time.Now()` each server would reach a different value for this.

That's why for non-deterministic values like this we set them at the RPC layer:
https://github.com/hashicorp/nomad/blob/1ef99f05364b7d3739befa6a789f0d55b2314dcf/nomad/job_endpoint.go#L342-L348

So we will need to do the same for the `Deregister` method.